app/services/event_generator/orangelight_event_generator.rb

```diff
@@ -32,7 +32,9 @@ def record_member_updated(record)
     end
 
     def valid?(record)
-      record.is_a? Coin
+      return true if record.is_a?(Coin) && record.decorate.parents.present?
+      Rails.logger.error("#{Time.current} - #{record.title.first} with id: #{record.id} has no parent numismatic issue and cannot be synced to OL")
```

Comment:
Because of the event generators are run in a [queue](https://github.com/pulibrary/figgy/blob/master/app/services/event_generator.rb#L14-L19), this error will be logged for many resource types that create message events. (ScannedResources, Geo resources, Ephemera, etc..). I don't think it's necessary or even desired to log in a `valid?` method. I would remove unless there's a good argument for keeping it. And then it should only be called when the record is a Coin.

Reply:
Thanks for that. I did notice that different validations run but didn't review it the other way back for example validating a different resource. It would still go through the coin validation and log the error. I would like to log it but it doesn't bother me to remove it.

Reply:
another thought I had following the event generator was not to generate an event at all when the valid? is false